package/ports_daemon/ports_daemon/src/ports_daemon.c

```diff
@@ -42,22 +43,33 @@ static int parse_options(int argc, char *argv[])
 {
   enum {
     OPT_ID_DEBUG = 1,
+    OPT_ID_CAN_ENABLE = 2,
   };
 
   const struct option long_opts[] = {
     {"debug", no_argument, 0, OPT_ID_DEBUG},
+    {"can-enable", no_argument, 0, OPT_ID_CAN_ENABLE},
     {0, 0, 0, 0},
   };
 
   int opt;
 
   while ((opt = getopt_long(argc, argv, "", long_opts, NULL)) != -1) {
+    // end of options
+    if (opt == -1) {
```

Comment:
Do we need this?  It should be covered by the check/assignment in the while loop.

Reply:
ah, good point; i missed the check in the while loop